### ISSUE TITLE
Rule request: `require-valid-form-groups` (or similar)

### ISSUE THREAD

**Issue by MelSumner** [ember-template-lint/ember-template-lint#1359]
Requesting a new rule that requires appropriate semantics for grouped form controls. Correctly grouped form controls will take one of two approaches: 

1. use `<fieldset>` + `<legend>` (preferred)
2. associate controls using WAI-ARIA (also acceptable)

## Allowed
 
```html
<fieldset>
  <legend>Preferred Mascot Version</legend>
  <div>
      <label for="radio-001">Chicago Zoey</label>
      <input id="radio-001" type="radio" name="prefMascot-Zoey" value="chicago zoey" />
  </div>
  <div>
      <label for="radio-002">Office Hours Tomster</label>
      <input id="radio-002" type="radio" name="prefMascot-OfficeHoursTomster" value="office hours tomster" />
  </div>
  <div>
      <label for="radio-003">A11y Zoey</label>
      <input id="radio-003" type="radio" name="prefMascot-Zoey" value="a11y zoey" />
  </div>
</fieldset>
```

```html
<div role="group" aria-labelledby="preferred-mascot-heading">
  <div id="preferred-mascot-heading">Preferred Mascot Version</legend>
  <label for="radio-001">Chicago Zoey</label>
  <input id="radio-001" type="radio" name="prefMascot-Zoey" value="chicago zoey" />
  <label for="radio-002">Office Hours Tomster</label>
  <input id="radio-002" type="radio" name="prefMascot-OfficeHoursTomster" value="office hours tomster" />
  <label for="radio-003">A11y Zoey</label>
  <input id="radio-003" type="radio" name="prefMascot-Zoey" value="a11y zoey" />
</div>
```
## Disallowed

```html
<div>
  <label for="radio-001">Chicago Zoey</label>
  <input id="radio-001" type="radio" name="prefMascot-Zoey" value="chicago zoey" />
  <label for="radio-002">Office Hours Tomster</label>
  <input id="radio-002" type="radio" name="prefMascot-OfficeHoursTomster" value="office hours tomster" />
  <label for="radio-003">A11y Zoey</label>
  <input id="radio-003" type="radio" name="prefMascot-Zoey" value="a11y zoey" />
</div>
```
    
## References
- https://www.w3.org/WAI/tutorials/forms/grouping/
- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset

**Comment by rwjblue** [comment]
Hmm, this seems to be conflating two things:

* should use a fieldset when you have multiple related form elements (e.g. radio or checkboxes)
* each fieldset should have a legend

I'm not really sure how to enforce the first very well. Certainly having multiple radio buttons warrants a fieldset, but it doesn't seem obvious that anytime you have checkboxes that they are related.

Also, we'd need to ensure that we account for `role="group"` as a replacement for requiring a fieldset.

@MelSumner - Can you update the description to include any references that a rule implementer might need?

**Comment by MelSumner** [comment]
@rwjblue yup, updated. 👍

**Comment by rajasegar** [comment]
@MelSumner there were some typos in your second  example for allowed, I think that should be
```hbs
<div role="group" aria-labelledby="preferred-mascot-heading">
  <div id="preferred-mascot-heading">Preferred Mascot Version</div>
  <label for="radio-001">Chicago Zoey</label>
  <input id="radio-001" type="radio" name="prefMascot-Zoey" value="chicago zoey" />
  <label for="radio-002">Office Hours Tomster</label>
  <input id="radio-002" type="radio" name="prefMascot-OfficeHoursTomster" value="office hours tomster" />
  <label for="radio-003">A11y Zoey</label>
  <input id="radio-003" type="radio" name="prefMascot-Zoey" value="a11y zoey" />
</div>
```

**Comment by rajasegar** [comment]
I am picking this one,  can you please assign it to me.
@MelSumner I have some doubts 
- Where we should throw the lint errors for the form elements or the invalid form group control?
- How will we identify the form groups say for example is this considered to be a valid form group?
```hbs
<div>
  <label for="radio-001">Chicago Zoey</label>
</div>
```
or this one
```hbs
<div>
<input id="radio-001" type="radio" name="prefMascot-Zoey" value="chicago zoey" />
</div>
```
Is it something like a input and label elements side-by-side?
Because in the current implementation (which is actually quite primitive) I am having only one form element within a group and throwing errors for the same, since having two elements throw two errors simultaneously...
@rwjblue Need your inputs also for the same.
